### PR TITLE
feat: add lock status to submission lock responses

### PR DESCRIPTION
**Pull request by nsprenkle**
**TL;DR -** Add `lock_status` to submission lock responses.

JIRA: Supporting work in [AU-366](https://openedx.atlassian.net/browse/AU-366)
Linked PR: TBD

**What changed?**

- Added `lock_status` to `SubmissionLockSerializer ` and all submission lock responses.
- Updated response for absent/expired submission lock. Returns `lock_status: 'unlocked'` instead of empty dict.
- Updated docstrings.

**Developer Checklist**
n/a

**Testing Instructions**

Exercise the following endpoints and test-cases (NOTE - all requests require a valid `X-CSRFToken` header).

Check submission lock:
- `POST {{protocol}}://{{lms_url}}/courses/{{course_id}}/xblock/{{block_id}}/handler/check_submission_lock`

Claim a submission lock:
- `POST {{protocol}}://{{lms_url}}/courses/{{course_id}}/xblock/{{block_id}}/handler/claim_submission_lock`

Delete submission lock:
- `POST {{protocol}}://{{lms_url}}/courses/{{course_id}}/xblock/{{block_id}}/handler/delete_submission_lock`

Expected return values:
```
Success: HTTP 200 {
    "submission_uuid": <submisison_uuid or empty>,
    "owner_id": <anonymous_user_id or empty>,
    "created_at": <timestamp or empty>,
    "lock_status": <"unlocked", "in-progress", or "locked">
}

Contested lock: HTTP 403 {
    "error": <error message>
}
```
**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

File: openassessment/staffgrader/models/submission_lock.py [hunk 1]
```diff
@@ -91,6 +90,6 @@ def clear_submission_lock(cls, submission_uuid, user_id):
 
         # Only the owner can clear the lock
         if current_lock.owner_id != user_id:
-            raise SubmissionLockContestedError(_("Unable to clear submission lock"))
+            raise SubmissionLockContestedError("Unable to clear submission lock")
```
Comment (mattcarter) [inline openassessment/staffgrader/models/submission_lock.py:93]: Not translating?

File: openassessment/staffgrader/models/submission_lock.py [hunk 2]
```diff
@@ -63,7 +62,7 @@ def claim_submission_lock(cls, submission_uuid, user_id):
         # If there's already an active lock, raise an error
         # Unless the lock owner is trying to reacquire a lock, which is allowed
         if current_lock and current_lock.is_active and current_lock.owner_id != user_id:
-            raise SubmissionLockContestedError(_("Submission already locked"))
+            raise SubmissionLockContestedError("Submission already locked")
```
Comment (mattcarter) [inline openassessment/staffgrader/models/submission_lock.py:65]: Not translating?

**Review by mattcarter — COMMENTED** [review]
Generally, coo, coo, coo.  Question about translations for strings that look like user communication.

File: openassessment/staffgrader/models/submission_lock.py (hunk 1 above)
Comment (nsprenkle) [reply openassessment/staffgrader/models/submission_lock.py:93]: @muselesscreator , 📢 question about error-handling. I'm not sure where it makes sense to translate error messages: on the backend (so the frontend can just surface the message) or the frontend (so the backend message can be used for comparison/logic). Thoughts?

Edit: From discussion with @mattcarter, thinking it might make sense to provide both an error code (for comparison on the frontend) and pre-translated error message (for easy pass-through). Something like:

```
{ 'error_code': 'err_lock_contested', 'error_msg': (translatable)"Unable to clear submission lock" }
```

File: openassessment/staffgrader/tests/test_serializers.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+        # create a test lock for test_submission_id by test_user_id_1
+        self.test_submission_lock = SubmissionGradingLock.claim_submission_lock(
+            self.test_submission_id,
+            self.test_user_id_1,
+        )
+
+    def test_empty(self):
+        """ Serialization with an empty object returns lock_status of 'unlocked' """
+        context = {'user_id': self.test_user_id_1, 'submission_uuid': self.test_submission_id}
```
Comment (jansenk) [inline openassessment/staffgrader/tests/test_serializers.py:39]: Why is submission_uuid included in context?

File: openassessment/staffgrader/tests/test_serializers.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        assert SubmissionLockSerializer({}, context=context).data == expected_output
+
+    def test_serialize_inactive_lock(self):
+        """ An inactive lock should serialize with lock_status of 'unlocked'. Other fields may or may not be passed """
+        self.test_submission_lock.created_at = self.test_submission_lock.created_at - (
+            SubmissionGradingLock.TIMEOUT + timedelta(hours=1)
+        )
+        self.test_submission_lock.save()
+
+        context = {'user_id': self.test_user_id_1, 'submission_uuid': self.test_submission_id}
```
Comment (jansenk) [inline openassessment/staffgrader/tests/test_serializers.py:50]: here as well?

File: openassessment/staffgrader/errors/submission_lock.py [hunk 5]
```diff
@@ -7,3 +7,7 @@ class SubmissionLockContestedError(Exception):
     """
     Error indicating trying to modify a lock that the user does not have access to modify.
     """
+    error_code = 'ERR_LOCK_CONTESTED'
+
+    def __str__(self):
```
Comment (jansenk) [inline openassessment/staffgrader/errors/submission_lock.py:12]: I would maybe not have this be the __str__ for this exception? Maybe introduce a new method like "get_error_code()" or something? Any thoughts on that?

**Review by jansenk — APPROVED** [review]
Looks good, some small questions

File: openassessment/staffgrader/tests/test_serializers.py (hunk 4 above)
Comment (nsprenkle) [reply openassessment/staffgrader/tests/test_serializers.py:50]: I had debated using this so you will always get a submission ID even if there isn't an active lock, but never implemented that logic. Will remove.

File: openassessment/staffgrader/errors/submission_lock.py (hunk 5 above)
Comment (nsprenkle) [reply openassessment/staffgrader/errors/submission_lock.py:12]: I had played with a few option. I did this just so we can do the same `str(ex)` for any exception type but it is a bit hacky.
